scalafix-core/shared/src/main/scala/scalafix/internal/rule/DottyKeywords.scala

```diff
@@ -14,8 +14,8 @@ case object DottyKeywords extends Rule("DottyKeywords") {
   override def fix(ctx: RuleCtx): Patch =
     ctx.tree.collect {
       case name @ Name("enum") =>
-        ctx.replaceTree(name, s"`enum`")
+        ctx.replaceTree(name, s"`enum`").atomic
       case name @ Name("inline") if !name.parents.exists(_.is[Mod.Annot]) =>
-        ctx.replaceTree(name, s"`inline`")
+        ctx.replaceTree(name, s"`inline`").atomic
     }.asPatch
```

Comment:
@marcelocenerine Surely I can do the `.atomic`s right after `.asPatch` instead of in the match cases?

Reply:
That's a very good question. The answer is 'no'. The single patch returned by `Rule.fix` groups all the rewrites for the entire source file. If we make it atomic, then all the rewrites for a given file would be rolled back despite users might wanted to suppress only one particular occurrence of code that triggers the rule.
Instead, what we need to do is to make sure that each occurrence of a rewrite is atomic. For example, if you wanted to rewrite usage of infix operators you would have to combine multiple patches, like:

input:
```scala
obj doSomething 1
```

expected output:
```scala
obj.doSomething(1)
```

rule (probably incomplete):
```
ctx.removeTokens(tokenList.trailingSpaces(op))
ctx.removeTokens(tokenList.leadingSpaces(op))
ctx.addLeft(op, ".")
ctx.addRight(op, "(")
ctx.addRight(args, ")")
```

In this case, if users wanted to suppress this rewrite for one particular occurrence of the infix operator in their code, scalafix should suppress all the 5 patches together.

We need to run the tests and check what rules are not emiting atomic top level patches for rewrites. Then for each rule we need to see what makes sense to be grouped in an atomic patch and put the `.atomic` call there.

Reply:
@olafurpg btw, I think such rule 👆would be a good addition 😄

Reply:
What rule exactly do you have in mind @marcelocenerine ? I'm wondering if it makes sense to automatically insert `.atomic` for every value in `.asPatch`

Reply:
@olafurpg it would be probably ok for most cases where `.asPatch` is used at the end of `.collect`. But maybe for some uncommon use cases it could be as dangerous as https://github.com/scalacenter/scalafix/pull/706#discussion_r182087220 since an individual `TokenPatch` would become "suppress-able". Do you think that is a good compromise? I was thinking the same for `RuleCtx.replaceTree` btw.

Reply:
I noticed that `.asPatch` is used in several places in the scalafix repo where automatically inserting `.atomic` to each element would not make sense. I think it's best to force people to be explicit with `.map(_.atomic)` with potentially a `.mapAtomic` extension method for convenience.